### ISSUE TITLE
Maximum force does not read correctly

### ISSUE THREAD

**Issue by tw2469** [Ionizing/rsgrad#13]
[OUTCAR.txt](https://github.com/user-attachments/files/16459870/OUTCAR.txt)
[POSCAR.txt](https://github.com/user-attachments/files/16459886/POSCAR.txt)

I think cgrad gives the correct force:
```
   1 Energy:-1268.30125  Log|dE|: 3.1  SCF: 51  Avf|F|: 0.048  Max|F|: 0.151  AtomInd:128z  Time: 48.67
   2 Energy:-1268.30533  Log|dE|:-2.4  SCF:  6  Avf|F|: 0.036  Max|F|: 0.215  AtomInd:132z  Time:  4.99
   3 Energy:-1268.30507  Log|dE|:-3.6  SCF: 10  Avf|F|: 0.041  Max|F|: 0.287  AtomInd:132z  Time:  8.26
   4 Energy:-1268.30770  Log|dE|:-2.6  SCF:  9  Avf|F|: 0.036  Max|F|: 0.123  AtomInd: 84z  Time:  7.05
   5 Energy:-1268.31102  Log|dE|:-2.5  SCF:  6  Avf|F|: 0.045  Max|F|: 0.168  AtomInd:138x  Time:  5.20
```

I wrote a quick script to verify this:
```
     1  -1268.23086765  10)      0.151413
     2  -1268.23502776  13)      0.214847
     3  -1268.23461562  14)      0.286599
     4  -1268.23711316  14)      0.122645
     5  -1268.23991888  9)       0.168318
```

This is the results from rsgrad rlx:
```
  #Step  TOTEN_z/eV LgdE   Fmax #SCF Time/m Mag/muB
      1 -1268.23087  3.1  0.071   51  50.06   NoMag
      2 -1268.23503 -2.4  0.034    6   6.10   NoMag
      3 -1268.23462 -3.4  0.048   10   9.37   NoMag
      4 -1268.23711 -2.6  0.059    9   8.16   NoMag
      5 -1268.23992 -2.6  0.081    6   6.31   NoMag
```

**Comment by Ionizing** [comment]
This is a `rsgrad` bug raised by the incorrect `nions`:

Here

https://github.com/Ionizing/rsgrad/blob/0eb3249dff9557cf3715eb50a95598ea42c4d50e/src/vasp_parsers/outcar.rs#L763

```rust
let len = self._data.len();
let dynamics = self._constraints.as_ref()
                   .unwrap_or(&vec![[true; 3]; len])
...
```

should be

```rust
let len = self._data[0].positions.len();
let dynamics = self._constraints.as_ref()
                    .unwrap_or(&vec![[true; 3]; len])
                    .iter()
...
```

Patch with the modification above, the `rsgrad` should produces correct Fmax information.

```
[2024-08-02T03:53:15Z INFO  rsgrad::commands::rlx] Parsing file "/Users/ionizing/Downloads/OUTCAR (1).txt" and "./POSCAR"
[2024-08-02T03:53:15Z WARN  rsgrad::commands::rlx] Reading constraints from POSCAR file "./POSCAR" failed
  #Step  TOTEN_z/eV LgdE   Fmax #SCF Time/m Mag/muB
      1 -1268.23087  3.1  0.151   51  50.06   NoMag
      2 -1268.23503 -2.4  0.215    6   6.10   NoMag
      3 -1268.23462 -3.4  0.287   10   9.37   NoMag
      4 -1268.23711 -2.6  0.123    9   8.16   NoMag
      5 -1268.23992 -2.6  0.168    6   6.31   NoMag
      6 -1268.24152 -2.8  0.109   11  10.26   NoMag
      7 -1268.24196 -3.4  0.124   15  12.90   NoMag
      8 -1268.24466 -2.6  0.118   11   9.64   NoMag
      9 -1268.24572 -3.0  0.208    7   7.07   NoMag
     10 -1268.24759 -2.7  0.109   10   9.49   NoMag
     11 -1268.24785 -3.6  0.150   10   9.88   NoMag
     12 -1268.25007 -2.7  0.088   13  11.87   NoMag
     13 -1268.25065 -3.2  0.200   14  12.22   NoMag
     14 -1268.25239 -2.8  0.080   14  12.14   NoMag
     15 -1268.25229 -4.0  0.134    9   8.08   NoMag
     16 -1268.25568 -2.5  0.096    6   6.26   NoMag
     17 -1268.25506 -3.2  0.105   10   9.30   NoMag
     18 -1268.25517 -3.9  0.210   10   9.14   NoMag
     19 -1268.25636 -2.9  0.145   15  12.85   NoMag
[2024-08-02T03:53:15Z INFO  rsgrad] Time used: 7.594309ms
```

**Comment by Ionizing** [comment]
Fixed via 7a0d159964d2b3c950c06723ea73e908248d2024
